src/databricks/labs/ucx/source_code/dependencies.py

```diff
@@ -104,16 +105,31 @@ def _load_source(self, object_info: ObjectInfo) -> str:
 class DependencyResolver:
     def __init__(self, whitelist: Whitelist | None = None):
         self._whitelist = Whitelist() if whitelist is None else whitelist
+        self._advices: list[Advice] = []
 
     def resolve(self, dependency: Dependency) -> Dependency | None:
         if dependency.type == ObjectType.NOTEBOOK:
             return dependency
         compatibility = self._whitelist.compatibility(dependency.path)
         # TODO attach compatibility to dependency, see https://github.com/databrickslabs/ucx/issues/1382
         if compatibility is not None:
+            if compatibility == UCCompatibility.NONE:
+                self._advices.append(
+                    Deprecation(
+                        code="dependency-check",
+                        message=f"Use of dependency {dependency.path} is deprecated",
+                        start_line=0,
```

Comment:
How can we modify it to highlight the exact code range where deprecated import happened?...

Reply:
Two things here. I think that it would be better to not report during graph building, but do a walk after. But add more information based on object type. So file, token etc. hence I wanted two phases to this. I talked briefly with @ericvergnaud about that.

Reply:
Exactly. But that info is not available right now. When the dependency is constructed, I need to store the origin information in the ObjectInfo. Then the Resolver can use it. Probably with another traversal.